discovery/marathon/marathon.go

```diff
@@ -298,15 +315,26 @@ func targetsForApp(app *App) []model.LabelSet {
 		if len(t.Ports) == 0 {
 			continue
 		}
-		target := targetForTask(&t)
-		targets = append(targets, model.LabelSet{
-			model.AddressLabel: model.LabelValue(target),
-			taskLabel:          model.LabelValue(t.ID),
-		})
+		for i := 0; i < len(t.Ports); i++ {
+			targetAddress := targetForTask(&t, i)
+			target := model.LabelSet{
+				model.AddressLabel: model.LabelValue(targetAddress),
+				taskLabel:          model.LabelValue(t.ID),
+			}
+			for ln, lv := range app.PortDefinitions[i].Labels {
```

Comment:
Is it guaranteed that the API response will include `i` port definitions? I have the feeling we should wrap this in a conditional and ensure we can't throw out of bound panics here.